projects/sonar/src/app/user.service.ts

```diff
@@ -82,6 +91,34 @@ export class UserService {
     return false;
   }
 
+  /**
+   * Check is user as at least the given role.
+   * @param role Role to check.
+   * @returns True if user has the role.
+   */
+  is(role: string): boolean {
+    switch (role) {
+      case 'user': {
+        return this._user.is_user;
+      }
+      case 'publisher': {
+        return this._user.is_publisher;
+      }
+      case 'moderator': {
+        return this._user.is_user;
```

Comment:
really?

Reply:
oh shit.. But I will refactor this method because it's not well implemented.